### PR TITLE
PYTHON-2940 Fix spec tests that require DNSPython

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: test/test_uri_spec.py [hunk 1]
```diff
@@ -181,6 +182,12 @@ def create_tests(test_path):
                 scenario_def = json.load(scenario_stream)
 
             for testcase in scenario_def['tests']:
+                if (testcase['uri'].startswith("mongodb+srv") and not
+                _HAVE_DNSPYTHON):
```
Comment (ShaneHarvey) [inline test/test_uri_spec.py:186]: The code style is off here

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/test_uri_spec.py (hunk 1 above)
Comment (ShaneHarvey) [reply test/test_uri_spec.py:186]: The indentation is still off.

File: test/test_uri_spec.py [hunk 2]
```diff
@@ -181,6 +182,12 @@ def create_tests(test_path):
                 scenario_def = json.load(scenario_stream)
 
             for testcase in scenario_def['tests']:
+                if (testcase['uri'].startswith(SRV_SCHEME) and not
```
Comment (ShaneHarvey) [inline test/test_uri_spec.py:185]: Please move this skipping logic to the `run_scenario` method above and use self.skip(...) instead of print.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/test_uri_spec.py [hunk 3]
```diff
@@ -92,7 +93,10 @@ def run_scenario(self):
         compressors = (test.get('options') or {}).get('compressors', [])
         if 'snappy' in compressors and not _HAVE_SNAPPY:
             self.skipTest('This test needs the snappy module.')
-
+        if (test['uri'].startswith(SRV_SCHEME) and not
+                _HAVE_DNSPYTHON):
```
Comment (ShaneHarvey) [inline test/test_uri_spec.py:97]: This should be one line now.

File: test/test_uri_spec.py [hunk 4]
```diff
@@ -92,7 +93,10 @@ def run_scenario(self):
...
+            self.skipTest("Skipping test '%s' because you need DNSPython for "
+                  "mongodb+srv URIs" % dsc)
```
Comment (ShaneHarvey) [inline test/test_uri_spec.py:99]: Indentation.

File: test/test_uri_spec.py (hunk 3 above)
Comment (juliusgeo) [reply test/test_uri_spec.py]: Done.

File: test/test_uri_spec.py (hunk 4 above)
Comment (juliusgeo) [reply test/test_uri_spec.py]: Done.

**Pull request merged**